0.0.1/agent.py

```diff
@@ -1,63 +1,99 @@
-
+import logging
+import json
 import re
-from src.utils import fetch_coinbase, fetch_coingecko, get_recommended_token_allocations, get_near_account_balance
+import typing
+from nearai.agents.environment import Environment, ChatCompletionMessageToolCall
+from src.utils import (
+    fetch_coinbase,
+    fetch_coingecko,
+    get_recommended_token_allocations,
+    get_near_account_balance,
+)
 
 USD_PORTFOLIO_GOAL_REGEX = re.compile(r"portfolio:\s*(\d+)")
 USD_ALLOWANCE_GOAL_REGEX = re.compile(r"allowance:\s*(\d+)")
 
+
 class Agent:
 
-    def __init__(self, env):
+    def __init__(self, env: Environment):
         self.env = env
         self.growth_goal = None
         self.allowance_goal = None
         self.prices = None
         self.recommended_tokens = None
-        self.near_account_id = None
+        self.near_account_id = "ptke.near"
         self.near_account_balance = None
         tool_registry = self.env.get_tool_registry()
-        tool_registry.register_tool(self.recommend_token_allocations_to_swap_for_stablecoins)
+        tool_registry.register_tool(
+            self.recommend_token_allocations_to_swap_for_stablecoins
+        )
         tool_registry.register_tool(self.get_allowance_goal)
         tool_registry.register_tool(self.find_near_account_id)
 
-
     def find_growth_goal(self, chat_history):
         for message in reversed(chat_history):
-            match = USD_PORTFOLIO_GOAL_REGEX.match(message['content'])
-            if message['role'] == 'user' and match:
+            match = USD_PORTFOLIO_GOAL_REGEX.match(message["content"])
+            if message["role"] == "user" and match:
                 return match.group(1)
-        return ''
+        return ""
 
     def find_allowance_goal(self, chat_history):
         for message in reversed(chat_history):
-            match = USD_ALLOWANCE_GOAL_REGEX.match(message['content'])
-            if message['role'] == 'user' and match:
+            match = USD_ALLOWANCE_GOAL_REGEX.match(message["content"])
+            if message["role"] == "user" and match:
                 return match.group(1)
-        return ''
+        return ""
 
     def get_last_search_term(self, chat_history):
         for message in reversed(chat_history):
-           if message['role'] == 'user':
-               return message['content']
-        return ''
+            if message["role"] == "user":
+                return message["content"]
+        return ""
 
     def run(self):
         # A system message guides an agent to solve specific tasks.
-        prompt = {"role": "system", "content": "You are an assistant that helps people set goals for growth in the USD value of their crypto assets such that when that percentage in growth has been reached or surpassed, you look at their tokens and determine the tokens and quantities of each to swap for USDT stablecoins or USDC stablecoins"}
+        prompt = {
```

Comment:
Adjusted the prompt

Reply:
Interesting. So this extra statement about having access to auxiliary info is helpful in addition to explicitly calling `register_tool`?

Reply:
Indirectly. Without that line, it would refuse to parse and return this response instead
```
Assistant: I don't have any information about your NEAR account ID. I'm a text-based AI assistant and do not have access to your personal data or accounts. If you need to find your NEAR account ID, you can try checking your NEAR wallet or contacting the NEAR support team.
```

Reply:
Adjusted the prompt even further to constrain the range of outcomes and reduce hallucinations